### PR TITLE
to_app_option Err type should be SerializedBytesError

### PR DESCRIPTION
**Pull request by Connoropolous**
This is so that `to_app_option` matches the type that `#[hdk_entry]` inserts via the `holochain_serial!` macro

The solution, after I looked for it for hours myself, was proposed to me by @maackle here:
https://discord.com/channels/727959104777224303/728631961488326726/758446365910761494

this is a snip from holochain_serialized_bytes that shows that.
```
impl std::convert::TryFrom<&$t> for $crate::SerializedBytes {
                type Error = $crate::SerializedBytesError;
                fn try_from(t: &$t) -> std::result::Result<$crate::SerializedBytes, $crate::SerializedBytesError> {
                    match $crate::encode(t) {
                        Ok(v) => Ok($crate::SerializedBytes::from($crate::UnsafeBytes::from(v))),
                        Err(e) => Err($crate::SerializedBytesError::ToBytes(e.to_string())),
                    }
                }
            }

            impl std::convert::TryFrom<$t> for $crate::SerializedBytes {
                type Error = $crate::SerializedBytesError;
                fn try_from(t: $t) -> std::result::Result<$crate::SerializedBytes, $crate::SerializedBytesError> {
                    $crate::SerializedBytes::try_from(&t)
                }
            }

            impl std::convert::TryFrom<$crate::SerializedBytes> for $t {
                type Error = $crate::SerializedBytesError;
                fn try_from(sb: $crate::SerializedBytes) -> std::result::Result<$t, $crate::SerializedBytesError> {
                    match $crate::decode(sb.bytes()) {
                        Ok(v) => Ok(v),
                        Err(e) => Err($crate::SerializedBytesError::FromBytes(e.to_string())),
                    }
                }
            }
```

This means that if, in a Zome function with an `ExternResult`, we call `to_app_option()?` we can use the ? operator and it will compile.

### CODE REVIEW

**Comment by Connoropolous** [comment]
ping @thedavidmeister requesting your review :)

**Review by freesig — APPROVED** [review]
This looks good to me

**Comment by Connoropolous** [comment]
@freesig thanks. CI isn't running on this branch so that would make sense to enable, or run another way.

File: crates/zome_types/src/element.rs [hunk 1]
```diff
@@ -132,7 +132,9 @@ impl ElementEntry {
     /// anything other than ElementEntry::Present returns None
     /// a present entry that fails to deserialize cleanly is an error
     /// a present entry that deserializes cleanly is returned as the provided type A
-    pub fn to_app_option<A: TryFrom<SerializedBytes>>(&self) -> Result<Option<A>, A::Error> {
+    pub fn to_app_option<A: TryFrom<SerializedBytes, Error = SerializedBytesError>>(
+        &self,
+    ) -> Result<Option<A>, SerializedBytesError> {
```
Comment (Connoropolous) [inline crates/zome_types/src/element.rs:137]: I guess this could be more specific as `SerializedBytesError::FromError`?

**Comment by Connoropolous** [comment]
@freesig I just ran hc-merge-test on this locally and it passed, and I'm happy with it, so please merge it :)

**Comment by Connoropolous** [comment]
@maackle maybe could get you to merge this one, as I think I'm more in your timezone range than Tom's

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**